.github/workflows/release.yaml

```diff
@@ -0,0 +1,156 @@
+          fetch-depth: 0
+
+      - name: Create Pull Request
+        run: |
+          sed 's/A few sentences describing the overall goals of the pull request'"'"'s commits./Release v${{ env.version }}/' .github/pull_request_template.md > temp_release_pr_body.md
+          gh pr create --title "Release v${{ env.version }}" \
+                       --body-file "temp_release_pr_body.md" \
+                       --base main \
+                       --head ${{ env.branch_name }}
+          echo "Waiting for 30 seconds to ensure PR is registered..."
+          sleep 30
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Wait for PR to be merged
+        run: |
+          pr_number=$(gh pr list --base main --head ${{ env.branch_name }} --search "Release v${{ env.version }}" --json number --jq '.[0].number')
+          if [ -z "$pr_number" ]; then
+            echo "No PR found with the name Release v${{ env.version }}."
+            exit 1
+          fi
+
+          echo "Waiting for the PR to be merged. Please merge the PR manually."
+          timeout=60  # Timeout in minutes
+          while [ $timeout -gt 0 ]; do
+            pr_status=$(gh pr view "$pr_number" --json state --jq '.state')
+            if [ "$pr_status" = "MERGED" ]; then
+              echo "PR has been merged."
+              break
+            fi
+            echo "PR not merged yet. Sleeping for 1 minute..."
+            sleep 60
+            timeout=$((timeout - 1))
+          done
+          if [ $timeout -eq 0 ]; then
+            echo "Timeout reached. PR was not merged."
+            exit 1
+          fi
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  # TODO: attach built binaries to the release (if any like repctl, dellctl etc.)
+  create-release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    needs: create-pull-request
+    steps:
+      - name: Set up Git
+        run: |
+          git config --global user.name 'github-actions'
+          git config --global user.email 'github-actions@github.com'
+
+      - name: Checkout the code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Create new tag
+        run: |
+          git fetch --tags
+          git tag v${{ env.version }}
+          git push origin v${{ env.version }}
+
+      - name: Create Release
+        id: create_release
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: v${{ env.version }}
+          release_name: v${{ env.version }}
+          draft: true
+          prerelease: false
+          generate_release_notes: false
+          make_latest: true
+          body: |
+            ## Documentation
+            - [General Documentation](https://dell.github.io/csm-docs/docs/)
+            ## Change Log
+            See the [CHANGELOG](https://github.com/dell/csm/blob/main/CHANGELOG/CHANGELOG-${{ env.version }}.md) for more details.
```

Comment:
This would be a dead link if it were a patch release because the patch changes are always added to an existing changelog. See as an example: https://github.com/dell/csm/blob/main/CHANGELOG/CHANGELOG-1.11.0.md

Reply:
@shaynafinocchiaro, I updated the code with the patch release fix. 
Created a new PR with updates: https://github.com/dell/csm/pull/1649, Please review the same.

Closing this PR as this one is a protected branch.